otelcollector/shared/prometheus-ccp-config-merger.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"reflect"
+	"strings"
+
```

Comment:
could we keep the test that I added for this file here? https://github.com/Azure/prometheus-collector/blob/6c2e990f4615f19ad3d5a893cd7b67e9fbedf076/otelcollector/prometheuscollector/prometheus-ccp-config-merger_test.go

Reply:
added